application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,330 @@
+        }
+
+        return channelMonitor.statusMessage(guild);
+    }
+
+    /**
+     * Method for responding to 'onGuildMessageReceived' this will need to be replaced by a more
+     * appropriate method when the bot has more functionality.
+     * <p>
+     * Marks channels as busy when a user posts a message in a monitored channel that is currently
+     * free.
+     *
+     * @param event the generic event that includes the 'onGuildMessageReceived'.
+     */
+    @Override
+    public void onEvent(@NotNull GenericEvent event) {
+        if (event instanceof GuildMessageReceivedEvent guildEvent) {
+            if (guildEvent.isWebhookMessage() || guildEvent.getAuthor().isBot()) {
+                return;
+            }
+            if (!channelMonitor.isMonitoringChannel(guildEvent.getChannel().getIdLong())) {
+                logger.debug(
+                        "Channel is not being monitored, ignoring message received in {} from {}",
+                        guildEvent.getChannel().getName(), guildEvent.getAuthor());
+                return;
+            }
+            if (channelMonitor.isChannelBusy(guildEvent.getChannel().getIdLong())) {
+                logger.debug(
+                        "Channel status is currently busy, ignoring message received in {} from {}",
+                        guildEvent.getChannel().getName(), guildEvent.getAuthor());
+                return;
+            }
+            channelMonitor.setChannelBusy(guildEvent.getChannel().getIdLong(),
+                    guildEvent.getAuthor().getIdLong());
+            displayStatus(channelMonitor.getStatusChannelFor(guildEvent.getGuild()));
+            guildEvent.getMessage().reply(UserStrings.NEW_QUESTION.message()).queue();
+        }
+    }
+
+    private Optional<Message> getStatusMessageIn(@NotNull TextChannel channel) {
+        if (channelToStatusMessage.containsKey(channel.getIdLong())) {
+            Long id = channelToStatusMessage.get(channel.getIdLong());
+            if (id == null) {
+                return Optional.empty();
+            }
+            Message message = channel.getHistoryAround(id, 1).complete().getMessageById(id);
+            if (message == null) {
+                return Optional.empty();
+            }
+            return Optional.of(message);
+        }
+        return findExistingStatusMessage(channel);
+    }
+
+    private Optional<Message> findExistingStatusMessage(@NotNull TextChannel channel) {
+        // will only run when bots starts, afterwards its stored in a map
+        Optional<Message> result = channel.getHistory()
+            .retrievePast(100)
+            .map(history -> history.stream()
+                .filter(message -> !message.getEmbeds().isEmpty())
+                .filter(message -> message.getAuthor().equals(channel.getJDA().getSelfUser()))
+                // .filter(message -> STATUS_TITLE.equals(message.getEmbeds().get(0).getTitle()))
```

Comment:
added a `// FIXME`

this is just to prevent the bot from overriding embeds placed by other commands in the same channel as the status messages
(which probably shouldnt be done anyway)

.... i swear that i already commented on this topic, has this point been reviewed twice?

fixed

Reply:
fixed